Samples/WindowsCopilotRuntime/cs-wpf-sparse/WCRforWPF/App.xaml.cs

```diff
@@ -0,0 +1,197 @@
+            CLSCTX_RESERVED1 = 0x40,
+            CLSCTX_RESERVED2 = 0x80,
+            CLSCTX_RESERVED3 = 0x100,
+            CLSCTX_RESERVED4 = 0x200,
+            CLSCTX_NO_CODE_DOWNLOAD = 0x400,
+            CLSCTX_RESERVED5 = 0x800,
+            CLSCTX_NO_CUSTOM_MARSHAL = 0x1000,
+            CLSCTX_ENABLE_CODE_DOWNLOAD = 0x2000,
+            CLSCTX_NO_FAILURE_LOG = 0x4000,
+            CLSCTX_DISABLE_AAA = 0x8000,
+            CLSCTX_ENABLE_AAA = 0x10000,
+            CLSCTX_FROM_DEFAULT_CONTEXT = 0x20000,
+            CLSCTX_ACTIVATE_32_BIT_SERVER = 0x40000,
+            CLSCTX_ACTIVATE_64_BIT_SERVER = 0x80000,
+            CLSCTX_INPROC = CLSCTX_INPROC_SERVER | CLSCTX_INPROC_HANDLER,
+            CLSCTX_SERVER = CLSCTX_INPROC_SERVER | CLSCTX_LOCAL_SERVER | CLSCTX_REMOTE_SERVER,
+            CLSCTX_ALL = CLSCTX_SERVER | CLSCTX_INPROC_HANDLER
+        }
+    }
+
+    public App()
+    {
+    }
+
+    // Originally StartupUri="MainWindow.xaml" was used instead of Startup="OnStartUp" in
+    // the App.xaml Application properties and the startup code lived in MainWindow.xaml.cs
+    // But this design presents a flaw where the unpackaged process will show itself before
+    // the package process is able to activate. 
+    //
+    // This could be mitigated using this.hide on the MainWindows startup but the unpackaged
+    // process still showed itself for a split second.
+    //
+    // Using OnStartUp and intializating the MainWindow once we know we are in the packaged process
+    // proved to be a proper solution to avoid this issue.
+    private void OnStartUp(Object sender, StartupEventArgs e)
+    {
+        RestartWithIdentityIfNecessary();
+        InitializeComponent();
+    }
+
+    private async Task RestartWithIdentityIfNecessary()
+    {
+        // If we are in the packaged process, then present the MainWindow
+        if (IsPackagedProcess())
+        {
+            var mainWindow = new MainWindow();
+            MainWindow.Show(); // Show the main window
+            return;
+        }
+
+        // If we are still unpackaged process, then register the MSIX and ActivateApplication
+        Task install = RegisterSparsePackage();
+        await install;
+        RunWithIdentity();
+        Environment.Exit(0);
+    }
+
+    private async Task RegisterSparsePackage()
+    {
+        // We expect the MSIX to be in the same directory as the exe. 
+        string exePath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+        string externalLocation = exePath;
+        string sparsePkgPath = exePath + "\\WCRforWPFSparse.msix";
+
+        Uri externalUri = new Uri(externalLocation);
+        Uri packageUri = new Uri(sparsePkgPath);
+
+        Console.WriteLine("exe Location {0}", externalLocation);
+        Console.WriteLine("msix Address {0}", sparsePkgPath);
+
+        Console.WriteLine("  exe Uri {0}", externalUri);
+        Console.WriteLine("  msix Uri {0}", packageUri);
```

Comment:
As a non-console process, do these Console.WriteLine outputs go anywhere?

Reply:
Not really. I can remove them